homeassistant/components/lektrico/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""The Lektrico Charging Station integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from aiohttp import ClientSession
+from lektricowifi import Device, DeviceConnectionError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CHARGERS_PLATFORMS, DOMAIN, LB_DEVICES_PLATFORMS
+from .coordinator import LektricoDeviceDataUpdateCoordinator
+
+PARALLEL_UPDATES = 1
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Lektrico Charging Station from a config entry."""
+    session = async_get_clientsession(hass)
+
+    try:
+        coordinator = await create_coordinator(hass, entry, session)
+    except DeviceConnectionError as lek_ex:
+        raise ConfigEntryNotReady(lek_ex) from lek_ex
```

Comment:
Hello, 
Thank you for your comment.
When HomeAssistant boots and the device is offline, ConfigEntryNotReady  will show in frontend "Failed setup, will retry". And when the device comes online, the device will be initialized.
(if we put IntegrationError or ConfigEntryError, it will only show in frontend "Failed to set up" and won't try again).
Is there a better solution for this scenario  (HomeAssistant starts and the device is temporary offline) ?
Thank you for your help.